src/main.c

```diff
@@ -204,24 +205,47 @@ static const struct xdg_wm_base_listener xdg_wm_base_listener = {
 	.ping = xdg_wm_base_ping,
 };
 
-static void handle_dmabuf_format(void *data,
-		struct zwp_linux_dmabuf_v1 *zwp_linux_dmabuf, uint32_t format)
+static void handle_format_table(void *data,
+				struct zwp_linux_dmabuf_feedback_v1 *zwp_linux_dmabuf_feedback_v1,
+				int32_t fd, uint32_t size)
 {
 	(void)data;
-	(void)zwp_linux_dmabuf;
+	(void)zwp_linux_dmabuf_feedback_v1;
 
-	if (dmabuf_format != DRM_FORMAT_INVALID)
+	void* format_table = mmap(NULL, size, PROT_READ, MAP_PRIVATE, fd, 0);
+	if (format_table == MAP_FAILED)
 		return;
 
-	switch (format) {
-	case DRM_FORMAT_XRGB8888:
-	case DRM_FORMAT_XBGR8888:
-		dmabuf_format = format;
+	uint32_t offset = 0;
+	while (offset < size) {
+		uint32_t* format = format_table + offset;
```

Comment:
Pointer arithmetic on `void*` is illegal even though gcc allows it.

It would be better to define `format` as something like this:
```
struct format_table_entry {
    uint32_t format;
    uint32_t padding;
    uint64_t modifier;
} __attribute__((packed));
```

Then you can use something `for (int i = 0; i < size / sizeof(struct format_table_element); ++i)` to iterate through it.